### PR TITLE
feat: Polling rate to config flow

### PR DESCRIPTION
**Pull request by petretiandrea**

### CODE REVIEW

File: custom_components/tapo/translations/en.json [hunk 1]
```diff
@@ -10,10 +10,20 @@
...
-          "username": "Tapo App Username (email)"
+          "username": "Tapo App Username (email)",
+          "advanced_settings": "Advanced settings"
+        }
+      },
+      "advanced_config": {
+        "title": "Advanced configuration",
+        "description": "By customizing the polling rate unexpected behavior could happens. The device could even become unresponsive for a while.",
+        "data": {
+          "scan_interval": "Refresh rate in milliseconds"
```
Comment (steve-grb) [inline custom_components/tapo/translations/en.json:26]:
```suggestion
          "scan_interval": "Refresh rate in seconds"
```
I believe this is in seconds.  BTW I really appreciate you taking the time to implement this functionality. Can't wait to be released :D

**Comment by DJMalachite** [comment]
This will be great as i would like to see Realtime data :D

**Review by steve-grb — APPROVED** [review]
I already tested this locally and it seems to work great.

**Comment by KieranHare99** [comment]
Any update on when this will be approved?

**Comment by KieranHare99** [comment]
Just pushed this to my home assistant environment. Everything seems to be working. The only error I could see is once in the advanced options for setting the polling rate. The GUI specifies milliseconds but the text box actually uses seconds.

**Comment by profiluefter** [comment]
Hi, this patch is also important for my use case of triggering an automation if the button on the tapo is pressed manually.
I created a [fork](https://github.com/profiluefter/home-assistant-tapo-p100) which can be added in HACS if anyone else also wants to try this out before this PR is merged. I just applied this PR to the latest master (IntelliJ automagically resolved the conflicts) and also included the change from the review.

I can also confirm that everything seems to be working fine!

**Comment by davidsch1992** [comment]
Hello,

Where should I be able to adjust the polling rate? 
I cannot seem to find it. Do I need to re-add my Tapo devices for it to show up? 
Running HA with docker. 

Best Regards,

David

**Comment by profiluefter** [comment]
@davidsch1992 I think you have to re-add them and tick "Advanced settings" in the add-dialog. Not sure if there is a way to edit the value (aside from maybe some internal files).

**Comment by davidsch1992** [comment]
Hello,

It is working when you add new devices to specify the time interval. I tried 5 seconds, seem to work fine.

/David

**Pull request merged**